File: emarsys/src/main/java/com/emarsys/feature/InnerFeature.java
package com.emarsys.feature;

import com.emarsys.core.api.experimental.FlipperFeature;

public enum InnerFeature implements FlipperFeature {
    MOBILE_ENGAGE, PREDICT;

    @Override
    public String getName() {
        return "inner_feature_" + name().toLowerCase();
    }
}

File: emarsys/src/main/java/com/emarsys/feature/InnerFeature.kt
package com.emarsys.feature

import com.emarsys.core.api.experimental.FlipperFeature
import java.util.*

enum class InnerFeature : FlipperFeature {
    MOBILE_ENGAGE, PREDICT;

    override fun getName(): String {
        return "inner_feature_" + name.toLowerCase(Locale.getDefault())
    }

    companion object {
        fun safeValueOf(enumAsString: String): InnerFeature? {
            return try {
                valueOf(enumAsString)
            } catch (ignored: IllegalArgumentException) {
                null
            }
        }
    }
}